### ISSUE TITLE
Performance for GRPO on K-K dataset? Data Breach?

### ISSUE THREAD

**Issue by outongyiLv** [BytedTsinghua-SIA/Enigmata#8]
Hello everyone, I have a super serious problem for the dataset: verifiable_tasks/tasks/knights_and_knaves. I believe there is a data breach? Here is my settings:

I Used the Train Set: 3414 samples (926 easy, 1724 medium, 764 hard) for training.
I Used the Test Set: 600 samples (200 easy, 200 medium, 200 hard) for testing.

I used the functions written in verifiable_tasks/tasks/knights_and_knaves/verifier.py  as the score function to evaluate the score.

Here is my training curve for grpo, why acc for training become 100%？ and at the same time the evaluate acc on test dataset: easy, medium and hard all to 100%?  Could it be that there is something in the training set that directly relates to the test set???

I set the training-batch to 1024 and used the qwen-2.5-1.5b-instruct as the base model for GRPO training. Here is the training curve on training-data.

<img width="728" height="372" alt="Image" src="https://github.com/user-attachments/assets/175ffb08-95f0-4e88-9ea3-862e870b240a" />

And here is the three evaluation result during training:

<img width="657" height="351" alt="Image" src="https://github.com/user-attachments/assets/85bb1dde-6526-440d-a158-6329b0343c47" />

<img width="667" height="352" alt="Image" src="https://github.com/user-attachments/assets/cd64e975-e41d-4da6-86fe-c771c1bf237c" />

<img width="660" height="351" alt="Image" src="https://github.com/user-attachments/assets/c32ba552-f33e-425b-9535-7856b368bfb0" />

Or perhaps the training and test sets here aren't what I'm thinking; they're not inherently suitable for RLVR training?
Could someone kindly answer this for me? Thank you so much!
